specs/CriticalUpdateAvailable.md

```diff
@@ -0,0 +1,104 @@
+        RestartIfSelectedByUser();
+    }, null);
+}
+```
+
+## Win32 C++
+```cpp
+    wil::com_ptr<ICoreWebView2Environment> m_webViewEnvironment;
+    auto env10 = m_webViewEnvironment.try_query<ICoreWebView2Environment10>();
+    CHECK_FAILURE(env10->add_CriticalUpdateAvaliable(
+        Callback<ICoreWebView2CriticalUpdateAvaliableEventHandler>(
+            [this](ICoreWebView2Environment* sender, IUnknown* args) -> HRESULT
+            {
+                // Depending on your app experience, you may or may not
+                // want to prompt user to restart the app.
+                RestartIfSelectedByUser();
+                return S_OK;
+            })
+            .Get(),
+        nullptr));
+```
+
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+## Win32 C++
+
+```IDL
+interface ICoreWebView2Environment10;
+interface ICoreWebView2CriticalUpdateAvaliableEventHandler;
+
+[uuid(62cb67c6-b6a9-4209-8a12-72ca093b9547), object, pointer_default(unique)]
+interface ICoreWebView2CriticalUpdateAvaliableEventHandler : IUnknown {
+  /// Provides the event args for the corresponding event.  No event args exist
+  /// and the `args` parameter is set to `null`.
+  HRESULT Invoke([in] ICoreWebView2Environment* sender, [in] IUnknown* args);
+}
+
+[uuid(ef7632ec-d86e-46dd-9d59-e6ffb5c87878), object, pointer_default(unique)]
+interface ICoreWebView2Environment10 : IUnknown {
+  /// Add an event handler for the `CriticalUpdateAvaliable` event.
+  /// `CriticalUpdateAvaliable` event is raised when there is an urgent need to prompt the user 
+  /// to restart the WebView2 process to apply a particular configuration. The configuration 
+  /// is authored to include special attribute to indicate a payload as critical.
+  /// WebView2 team will author critical kill switch when there is a need to enable/disable 
+  /// certain features that’s causing WebView2 reliability or performance drop that’s impacting customers.
+  /// `CriticalUpdateAvaliable` will give developer the ability to prompt user for restart,
+  /// thus resolve in faster resolution time.
+  /// 
+  /// Critical Update is only applying payload; thus, version is not important. Simply prompt 
+  /// user to restart the app to get new payload applied.   
+  // MSOWNERS: xiaqu@microsoft.com
+  HRESULT add_CriticalUpdateAvaliable(
+      [in] ICoreWebView2CriticalUpdateAvaliableEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_CriticalUpdateAvaliable`.
+  // MSOWNERS: xiaqu@microsoft.com
+  HRESULT remove_CriticalUpdateAvaliable(
+      [in] EventRegistrationToken token);
+}
+```
+
+## .NET and WinRT
+
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2Environment
+    {
+        // ...
+        event Windows.Foundation.TypedEventHandler<CoreWebView2Environment, Object> CriticalUpdateAvaliable;
```

Comment:
This isn't actually related to updates. Its because we updated our ECS payload and need to restart to get different values for feature flags. What can we better name this event so it doesn't sound like its related to updating the runtime but also without mentioning new concepts of ECS / experimentation? Maybe something just about requiring a restart? @fabiorocha, @aluhrs13, @vickiez

Reply:
I somewhat disagree - From a consumer's perspective, an ECS payload is an update. Update doesn't _have_ to be new version/binaries.

Reply:
I would imagine most end devs are not aware of ECS payloads and would only think of 'update' as being new version/binaries. This event is never about new version/binaries. Maybe we could say like (Critical)RestartRequired/Requested?